lib/watch.js

```diff
@@ -37,18 +37,15 @@ function assertEncoding(encoding) {
 }
 
 function guard(fn) {
-  return function(arg, action) {
-    if (is.func(fn)) {
-      var ret = fn(arg, SKIP_FLAG);
-      if (ret && ret !== SKIP_FLAG) action();
+  if (is.func(fn))
+    return function(arg, action) {
+      if (fn(arg, false)) action();
     }
-    else if (is.regExp(fn)) {
+  if (is.regExp(fn))
+    return function(arg, action) {
       if (fn.test(arg)) action();
     }
-    else {
-      action();
-    }
-  }
+  return action;
```

Comment:
When I run `npm test` and I find this `action` is undefined :D

Reply:
I did the edit in the browser :sweat_smile:. Fixed now